common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataFlowTaskLaunchRequestAutoConfiguration.java

```diff
@@ -69,16 +68,37 @@ private Message dataflowTaskLaunchRequest(Message message) {
 		TaskLaunchRequestContext taskLaunchRequestContext = taskLaunchRequestContext(message);
 
 		DataFlowTaskLaunchRequest taskLaunchRequest = new DataFlowTaskLaunchRequest();
-		taskLaunchRequest.setCommandlineArguments(
-			taskLaunchRequestContext.mergeCommandLineArgs(taskLaunchRequestProperties));
+
+		List<String> evaluatedArgs = evaluateArgExpressions(message,
+			KeyValueListParser.parseCommaDelimitedKeyValuePairs(taskLaunchRequestProperties.getArgExpressions()));
```

Comment:
You need consider to parse expressions only once during initialization, not against every request.